storage/scratchpad/src/sparse_merkle/test_utils/proptest_helpers.rs

```diff
@@ -56,7 +70,25 @@ pub fn arb_smt_correctness_case() -> impl Strategy<Value = Vec<Action>> {
                                 .map(|updates| {
                                     updates
                                         .into_iter()
-                                        .map(|(k_idx, v)| (*k_idx.get(&keys), v.to_vec().into()))
+                                        .map(|(k_idx, v)| {
+                                            let key = *k_idx.get(&keys);
+                                            match v {
+                                                Op::Write(value) => {
+                                                    existing.insert(key);
+                                                    (key, Some(value.to_vec().into()))
+                                                }
+                                                Op::Delete(backup_write) => {
+                                                    if existing.contains(&key) {
+                                                        existing.remove(&key);
+                                                        println!("delete");
+                                                        (key, None)
+                                                    } else {
+                                                        existing.insert(key);
+                                                        (key, Some(backup_write.to_vec().into()))
+                                                    }
```

Comment:
Thought we don't care about deleting an non-existent key? This is..too much fun...

Reply:
yes, I didn't change this part but I can refine.